src/app/builders/components/BuilderRewards/BuilderRewards.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import { BuilderRewardDetails } from '@/app/collective-rewards/rewards'
+import { RewardCard, type RewardType } from './components/RewardCard'
+import { ClaimRewardsButton } from './buttons/ClaimRewardsButton'
+import { SeeRewardsHistoryButton } from './buttons/SeeRewardsHistoryButton'
+import { AdjustBackersRewardsButton } from './buttons/AdjustBackersRewardButton'
+import { Header } from '@/components/TypographyNew'
+
+interface BuilderRewardsProps extends BuilderRewardDetails {
+  className?: string
+  isMock?: boolean
+}
+
+// Configuration for each reward card with its specific props
+const REWARD_CARDS_CONFIG: Array<{
+  type: RewardType
+  showButton?: boolean
+  button?: React.ReactNode
+  specialLayout?: boolean
+}> = [
+  {
+    type: 'unclaimed',
+    showButton: true,
+    button: (
+      <ClaimRewardsButton
+        onClick={() => alert('Claim Rewards (placeholder - to be implemented in separate task)')}
+      />
+    ),
+  },
+  {
+    type: 'estimatedThisCycle',
+  },
+  {
+    type: 'lastCycle',
+  },
+  {
+    type: 'allTimeRewards',
+    showButton: true,
+    button: (
+      <SeeRewardsHistoryButton
+        onClick={() => alert('See Rewards history (placeholder - to be implemented in separate task)')}
+      />
+    ),
+  },
+  {
+    type: 'allTimeShare',
+    specialLayout: true,
```

Comment:
since we use the special layout for one column only, I'd suggest we don't configure it here and we just add another component after the loop.